// zxing-wasm/dist/es/writer/index.js
import { g as an, s as on, w as sn } from "../core-BYUmLXDu.js";
import { b as hn, e as gn, j as mn, j as yn, p as bn, k as wn } from "../core-BYUmLXDu.js";
var br = (() => {
  var G;
  var k = typeof document < "u" && ((G = document.currentScript) == null ? void 0 : G.tagName.toUpperCase()) === "SCRIPT" ? document.currentScript.src : void 0;
  return function(ee = {}) {
    var wr, u = ee, Tr, K, te = new Promise((r, e) => {
      Tr = r, K = e;
    }), ne = typeof window == "object", ae = typeof Bun < "u", sr = typeof WorkerGlobalScope < "u";
    typeof process == "object" && typeof process.versions == "object" && typeof process.versions.node == "string" && process.type != "renderer";
    var $r = Object.assign({}, u), A = "";
    function ie(r) {
      return u.locateFile ? u.locateFile(r, A) : A + r;
    }
    var Ar, ur;
    if (ne || sr || ae) {
      var cr;
      sr ? A = self.location.href : typeof document < "u" && ((cr = document.currentScript) === null || cr === void 0 ? void 0 : cr.tagName.toUpperCase()) === "SCRIPT" && (A = document.currentScript.src), k && (A = k), A.startsWith("blob:") ? A = "" : A = A.substr(0, A.replace(/[?#].*/, "").lastIndexOf("/") + 1), sr && (ur = (r) => {
        var e = new XMLHttpRequest();
        return e.open("GET", r, !1), e.responseType = "arraybuffer", e.send(null), new Uint8Array(e.response);
      }), Ar = async (r) => {
        var e = await fetch(r, {
          credentials: "same-origin"
        });
        if (e.ok)
          return e.arrayBuffer();
        throw new Error(e.status + " : " + e.url);
      };
    }
    u.print || console.log.bind(console);
    var H = u.printErr || console.error.bind(console);
    Object.assign(u, $r), $r = null, u.arguments && u.arguments, u.thisProgram && u.thisProgram;
    var q = u.wasmBinary, J, Er = !1, U, w, j, N, D, d, Rr, Cr;
    function Fr() {
      var r = J.buffer;
      u.HEAP8 = U = new Int8Array(r), u.HEAP16 = j = new Int16Array(r), u.HEAPU8 = w = new Uint8Array(r), u.HEAPU16 = N = new Uint16Array(r), u.HEAP32 = D = new Int32Array(r), u.HEAPU32 = d = new Uint32Array(r), u.HEAPF32 = Rr = new Float32Array(r), u.HEAPF64 = Cr = new Float64Array(r);
    }
    var Pr = [], Wr = [], kr = [];
    function oe() {
      if (u.preRun)
        for (typeof u.preRun == "function" && (u.preRun = [u.preRun]); u.preRun.length; )
          ce(u.preRun.shift());
      vr(Pr);
    }
    function se() {
      vr(Wr);
    }
    function ue() {
      if (u.postRun)
        for (typeof u.postRun == "function" && (u.postRun = [u.postRun]); u.postRun.length; )
          ve(u.postRun.shift());
      vr(kr);
    }
    function ce(r) {
      Pr.unshift(r);
    }
    function fe(r) {
      Wr.unshift(r);
    }
    function ve(r) {
      kr.unshift(r);
    }
    var x = 0, B = null;
    function le(r) {
      var e;
      x++, (e = u.monitorRunDependencies) === null || e === void 0 || e.call(u, x);
    }
    function de(r) {
      var e;
      if (x--, (e = u.monitorRunDependencies) === null || e === void 0 || e.call(u, x), x == 0 && B) {
        var t = B;
        B = null, t();
      }
    }
    function fr(r) {
      var e;
      (e = u.onAbort) === null || e === void 0 || e.call(u, r), r = "Aborted(" + r + ")", H(r), Er = !0, r += ". Build with -sASSERTIONS for more info.";
      var t = new WebAssembly.RuntimeError(r);
      throw K(t), t;
    }
    var _e = "data:application/octet-stream;base64,", Ur = (r) => r.startsWith(_e);
    function pe() {
      var r = "zxing_writer.wasm";
      return Ur(r) ? r : ie(r);
    }
    var Q;
    function he(r) {
      if (r == Q && q)
        return new Uint8Array(q);
      if (ur)
        return ur(r);
      throw "both async and sync fetching of the wasm failed";
    }
    async function ge(r) {
      if (!q)
        try {
          var e = await Ar(r);
          return new Uint8Array(e);
        } catch {
        }
      return he(r);
    }
    async function me(r, e) {
      try {
        var t = await ge(r), n = await WebAssembly.instantiate(t, e);
        return n;
      } catch (a) {
        H(`failed to asynchronously prepare wasm: ${a}`), fr(a);
      }
    }
    async function ye(r, e, t) {
      if (!r && typeof WebAssembly.instantiateStreaming == "function" && !Ur(e) && typeof fetch == "function")
        try {
          var n = fetch(e, {
            credentials: "same-origin"
          }), a = await WebAssembly.instantiateStreaming(n, t);
          return a;
        } catch (i) {
          H(`wasm streaming compile failed: ${i}`), H("falling back to ArrayBuffer instantiation");
        }
      return me(e, t);
    }
    function be() {
      return {
        a: Ot
      };
    }
    async function we() {
      var r;
      function e(i, o) {
        return y = i.exports, J = y.Y, Fr(), Hr = y.$, fe(y.Z), de(), y;
      }
      le();
      function t(i) {
        e(i.instance);
      }
      var n = be();
      if (u.instantiateWasm)
        try {
          return u.instantiateWasm(n, e);
        } catch (i) {
          H(`Module.instantiateWasm callback failed with error: ${i}`), K(i);
        }
      (r = Q) !== null && r !== void 0 || (Q = pe());
      try {
        var a = await ye(q, Q, n);
        return t(a), a;
      } catch (i) {
        K(i);
        return;
      }
    }
    var vr = (r) => {
      for (; r.length > 0; )
        r.shift()(u);
    };
    u.noExitRuntime;
    var g = (r) => Kr(r), m = () => qr(), Y = [], Te = (r) => {
      var e = new lr(r);
      return e.get_caught() || e.set_caught(!0), e.set_rethrown(!1), Y.push(e), Qr(r), zr(r);
    }, C = 0, $e = () => {
      p(0, 0);
      var r = Y.pop();
      Jr(r.excPtr), C = 0;
    };
    class lr {
      constructor(e) {
        this.excPtr = e, this.ptr = e - 24;
      }
      set_type(e) {
        d[this.ptr + 4 >> 2] = e;
      }
      get_type() {
        return d[this.ptr + 4 >> 2];
      }
      set_destructor(e) {
        d[this.ptr + 8 >> 2] = e;
      }
      get_destructor() {
        return d[this.ptr + 8 >> 2];
      }
      set_caught(e) {
        e = e ? 1 : 0, U[this.ptr + 12] = e;
      }
      get_caught() {
        return U[this.ptr + 12] != 0;
      }
      set_rethrown(e) {
        e = e ? 1 : 0, U[this.ptr + 13] = e;
      }
      get_rethrown() {
        return U[this.ptr + 13] != 0;
      }
      init(e, t) {
        this.set_adjusted_ptr(0), this.set_type(e), this.set_destructor(t);
      }
      set_adjusted_ptr(e) {
        d[this.ptr + 16 >> 2] = e;
      }
      get_adjusted_ptr() {
        return d[this.ptr + 16 >> 2];
      }
    }
    var Ae = (r) => {
      throw C || (C = r), C;
    }, z = (r) => Gr(r), dr = (r) => {
      var e = C;
      if (!e)
        return z(0), 0;
      var t = new lr(e);
      t.set_adjusted_ptr(e);
      var n = t.get_type();
      if (!n)
        return z(0), e;
      for (var a of r) {
        if (a === 0 || a === n)
          break;
        var i = t.ptr + 16;
        if (Yr(a, n, i))
          return z(a), e;
      }
      return z(n), e;
    }, Ee = () => dr([]), Re = (r) => dr([r]), Ce = (r, e) => dr([r, e]), Fe = () => {
      var r = Y.pop();
      r || fr("no exception to throw");
      var e = r.excPtr;
      throw r.get_rethrown() || (Y.push(r), r.set_rethrown(!0), r.set_caught(!1)), C = e, C;
    }, Pe = (r, e, t) => {
      var n = new lr(r);
      throw n.init(e, t), C = r, C;
    }, We = () => fr(""), rr = {}, _r = (r) => {
      for (; r.length; ) {
        var e = r.pop(), t = r.pop();
        t(e);
      }
    };
    function er(r) {
      return this.fromWireType(d[r >> 2]);
    }
    var O = {}, M = {}, tr = {}, Sr, xr = (r) => {
      throw new Sr(r);
    }, Mr = (r, e, t) => {
      r.forEach((s) => tr[s] = e);
      function n(s) {
        var c = t(s);
        c.length !== r.length && xr("Mismatched type converter count");
        for (var f = 0; f < r.length; ++f)
          E(r[f], c[f]);
      }
      var a = new Array(e.length), i = [], o = 0;
      e.forEach((s, c) => {
        M.hasOwnProperty(s) ? a[c] = M[s] : (i.push(s), O.hasOwnProperty(s) || (O[s] = []), O[s].push(() => {
          a[c] = M[s], ++o, o === i.length && n(a);
        }));
      }), i.length === 0 && n(a);
    }, ke = (r) => {
      var e = rr[r];
      delete rr[r];
      var t = e.rawConstructor, n = e.rawDestructor, a = e.fields, i = a.map((o) => o.getterReturnType).concat(a.map((o) => o.setterArgumentType));
      Mr([r], i, (o) => {
        var s = {};
        return a.forEach((c, f) => {
          var v = c.fieldName, l = o[f], h = c.getter, T = c.getterContext, S = o[f + a.length], L = c.setter, R = c.setterContext;
          s[v] = {
            read: (Z) => l.fromWireType(h(T, Z)),
            write: (Z, yr) => {
              var or = [];
              L(R, Z, S.toWireType(or, yr)), _r(or);
            }
          };
        }), [{
          name: e.name,
          fromWireType: (c) => {
            var f = {};
            for (var v in s)
              f[v] = s[v].read(c);
            return n(c), f;
          },
          toWireType: (c, f) => {
            for (var v in s)
              if (!(v in f))
                throw new TypeError(`Missing field: "${v}"`);
            var l = t();
            for (v in s)
              s[v].write(l, f[v]);
            return c !== null && c.push(n, l), l;
          },
          argPackAdvance: F,
          readValueFromPointer: er,
          destructorFunction: n
        }];
      });
    }, Ue = (r, e, t, n, a) => {
    }, Se = () => {
      for (var r = new Array(256), e = 0; e < 256; ++e)
        r[e] = String.fromCharCode(e);
      Ir = r;
    }, Ir, b = (r) => {
      for (var e = "", t = r; w[t]; )
        e += Ir[w[t++]];
      return e;
    }, jr, $ = (r) => {
      throw new jr(r);
    };
    function xe(r, e) {
      let t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : {};
      var n = e.name;
      if (r || $(`type "${n}" must have a positive integer typeid pointer`), M.hasOwnProperty(r)) {
        if (t.ignoreDuplicateRegistrations)
          return;
        $(`Cannot register type '${n}' twice`);
      }
      if (M[r] = e, delete tr[r], O.hasOwnProperty(r)) {
        var a = O[r];
        delete O[r], a.forEach((i) => i());
      }
    }
    function E(r, e) {
      let t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : {};
      return xe(r, e, t);
    }
    var F = 8, Me = (r, e, t, n) => {
      e = b(e), E(r, {
        name: e,
        fromWireType: function(a) {
          return !!a;
        },
        toWireType: function(a, i) {
          return i ? t : n;
        },
        argPackAdvance: F,
        readValueFromPointer: function(a) {
          return this.fromWireType(w[a]);
        },
        destructorFunction: null
      });
    }, pr = [], P = [], hr = (r) => {
      r > 9 && --P[r + 1] === 0 && (P[r] = void 0, pr.push(r));
    }, Ie = () => P.length / 2 - 5 - pr.length, je = () => {
      P.push(0, 1, void 0, 1, null, 1, !0, 1, !1, 1), u.count_emval_handles = Ie;
    }, I = {
      toValue: (r) => (r || $("Cannot use deleted val. handle = " + r), P[r]),
      toHandle: (r) => {
        switch (r) {
          case void 0:
            return 2;
          case null:
            return 4;
          case !0:
            return 6;
          case !1:
            return 8;
          default: {
            const e = pr.pop() || P.length;
            return P[e] = r, P[e + 1] = 1, e;
          }
        }
      }
    }, De = {
      name: "emscripten::val",
      fromWireType: (r) => {
        var e = I.toValue(r);
        return hr(r), e;
      },
      toWireType: (r, e) => I.toHandle(e),
      argPackAdvance: F,
      readValueFromPointer: er,
      destructorFunction: null
    }, Oe = (r) => E(r, De), Ve = (r, e, t) => {
      if (r[e].overloadTable === void 0) {
        var n = r[e];
        r[e] = function() {
          for (var a = arguments.length, i = new Array(a), o = 0; o < a; o++)
            i[o] = arguments[o];
          return r[e].overloadTable.hasOwnProperty(i.length) || $(`Function '${t}' called with an invalid number of arguments (${i.length}) - expects one of (${r[e].overloadTable})!`), r[e].overloadTable[i.length].apply(this, i);
        }, r[e].overloadTable = [], r[e].overloadTable[n.argCount] = n;
      }
    }, Dr = (r, e, t) => {
      u.hasOwnProperty(r) ? ((t === void 0 || u[r].overloadTable !== void 0 && u[r].overloadTable[t] !== void 0) && $(`Cannot register public name '${r}' twice`), Ve(u, r, r), u[r].overloadTable.hasOwnProperty(t) && $(`Cannot register multiple overloads of a function with the same number of arguments (${t})!`), u[r].overloadTable[t] = e) : (u[r] = e, u[r].argCount = t);
    }, He = (r, e, t) => {
      switch (e) {
        case 1:
          return t ? function(n) {
            return this.fromWireType(U[n]);
          } : function(n) {
            return this.fromWireType(w[n]);
          };
        case 2:
          return t ? function(n) {
            return this.fromWireType(j[n >> 1]);
          } : function(n) {
            return this.fromWireType(N[n >> 1]);
          };
        case 4:
          return t ? function(n) {
            return this.fromWireType(D[n >> 2]);
          } : function(n) {
            return this.fromWireType(d[n >> 2]);
          };
        default:
          throw new TypeError(`invalid integer width (${e}): ${r}`);
      }
    }, Ne = (r, e, t, n) => {
      e = b(e);
      function a() {
      }
      a.values = {}, E(r, {
        name: e,
        constructor: a,
        fromWireType: function(i) {
          return this.constructor.values[i];
        },
        toWireType: (i, o) => o.value,
        argPackAdvance: F,
        readValueFromPointer: He(e, t, n),
        destructorFunction: null
      }), Dr(e, a);
    }, nr = (r, e) => Object.defineProperty(e, "name", {
      value: r
    }), Or = (r) => {
      var e = Zr(r), t = b(e);
      return W(e), t;
    }, Vr = (r, e) => {
      var t = M[r];
      return t === void 0 && $(`${e} has unknown type ${Or(r)}`), t;
    }, Be = (r, e, t) => {
      var n = Vr(r, "enum");
      e = b(e);
      var a = n.constructor, i = Object.create(n.constructor.prototype, {
        value: {
          value: t
        },
        constructor: {
          value: nr(`${n.name}_${e}`, function() {
          })
        }
      });
      a.values[t] = i, a[e] = i;
    }, Xe = (r, e) => {
      switch (e) {
        case 4:
          return function(t) {
            return this.fromWireType(Rr[t >> 2]);
          };
        case 8:
          return function(t) {
            return this.fromWireType(Cr[t >> 3]);
          };
        default:
          throw new TypeError(`invalid float width (${e}): ${r}`);
      }
    }, Le = (r, e, t) => {
      e = b(e), E(r, {
        name: e,
        fromWireType: (n) => n,
        toWireType: (n, a) => a,
        argPackAdvance: F,
        readValueFromPointer: Xe(e, t),
        destructorFunction: null
      });
    };
    function Ze(r) {
      for (var e = 1; e < r.length; ++e)
        if (r[e] !== null && r[e].destructorFunction === void 0)
          return !0;
      return !1;
    }
    function Ge(r, e, t, n, a, i) {
      var o = e.length;
      o < 2 && $("argTypes array size mismatch! Must at least get return value and 'this' types!");
      var s = e[1] !== null && t !== null, c = Ze(e), f = e[0].name !== "void", v = o - 2, l = new Array(v), h = [], T = [], S = function() {
        T.length = 0;
        var L;
        h.length = s ? 2 : 1, h[0] = a, s && (L = e[1].toWireType(T, this), h[1] = L);
        for (var R = 0; R < v; ++R)
          l[R] = e[R + 2].toWireType(T, R < 0 || arguments.length <= R ? void 0 : arguments[R]), h.push(l[R]);
        var Z = n(...h);
        function yr(or) {
          if (c)
            _r(T);
          else
            for (var V = s ? 1 : 2; V < e.length; V++) {
              var nn = V === 1 ? L : l[V - 2];
              e[V].destructorFunction !== null && e[V].destructorFunction(nn);
            }
          if (f)
            return e[0].fromWireType(or);
        }
        return yr(Z);
      };
      return nr(r, S);
    }
    var Ke = (r, e) => {
      for (var t = [], n = 0; n < r; n++)
        t.push(d[e + n * 4 >> 2]);
      return t;
    }, qe = (r, e, t) => {
      u.hasOwnProperty(r) || xr("Replacing nonexistent public symbol"), u[r].overloadTable !== void 0 && t !== void 0 ? u[r].overloadTable[t] = e : (u[r] = e, u[r].argCount = t);
    }, Je = (r, e, t) => {
      r = r.replace(/p/g, "i");
      var n = u["dynCall_" + r];
      return n(e, ...t);
    }, ar = [], Hr, _ = (r) => {
      var e = ar[r];
      return e || (r >= ar.length && (ar.length = r + 1), ar[r] = e = Hr.get(r)), e;
    }, Qe = function(r, e) {
      let t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : [];
      if (r.includes("j"))
        return Je(r, e, t);
      var n = _(e)(...t);
      return n;
    }, Ye = (r, e) => function() {
      for (var t = arguments.length, n = new Array(t), a = 0; a < t; a++)
        n[a] = arguments[a];
      return Qe(r, e, n);
    }, X = (r, e) => {
      r = b(r);
      function t() {
        return r.includes("j") ? Ye(r, e) : _(e);
      }
      var n = t();
      return typeof n != "function" && $(`unknown function pointer with signature ${r}: ${e}`), n;
    }, ze = (r, e) => {
      var t = nr(e, function(n) {
        this.name = e, this.message = n;
        var a = new Error(n).stack;
        a !== void 0 && (this.stack = this.toString() + `
` + a.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return t.prototype = Object.create(r.prototype), t.prototype.constructor = t, t.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, t;
    }, Nr, rt = (r, e) => {
      var t = [], n = {};
      function a(i) {
        if (!n[i] && !M[i]) {
          if (tr[i]) {
            tr[i].forEach(a);
            return;
          }
          t.push(i), n[i] = !0;
        }
      }
      throw e.forEach(a), new Nr(`${r}: ` + t.map(Or).join([", "]));
    }, et = (r) => {
      r = r.trim();
      const e = r.indexOf("(");
      return e !== -1 ? r.substr(0, e) : r;
    }, tt = (r, e, t, n, a, i, o, s) => {
      var c = Ke(e, t);
      r = b(r), r = et(r), a = X(n, a), Dr(r, function() {
        rt(`Cannot call ${r} due to unbound types`, c);
      }, e - 1), Mr([], c, (f) => {
        var v = [f[0], null].concat(f.slice(1));
        return qe(r, Ge(r, v, null, a, i), e - 1), [];
      });
    }, nt = (r, e, t) => {
      switch (e) {
        case 1:
          return t ? (n) => U[n] : (n) => w[n];
        case 2:
          return t ? (n) => j[n >> 1] : (n) => N[n >> 1];
        case 4:
          return t ? (n) => D[n >> 2] : (n) => d[n >> 2];
        default:
          throw new TypeError(`invalid integer width (${e}): ${r}`);
      }
    }, at = (r, e, t, n, a) => {
      e = b(e);
      var i = (v) => v;
      if (n === 0) {
        var o = 32 - 8 * t;
        i = (v) => v << o >>> o;
      }
      var s = e.includes("unsigned"), c = (v, l) => {
      }, f;
      s ? f = function(v, l) {
        return c(l, this.name), l >>> 0;
      } : f = function(v, l) {
        return c(l, this.name), l;
      }, E(r, {
        name: e,
        fromWireType: i,
        toWireType: f,
        argPackAdvance: F,
        readValueFromPointer: nt(e, t, n !== 0),
        destructorFunction: null
      });
    }, it = (r, e, t) => {
      var n = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], a = n[e];
      function i(o) {
        var s = d[o >> 2], c = d[o + 4 >> 2];
        return new a(U.buffer, c, s);
      }
      t = b(t), E(r, {
        name: t,
        fromWireType: i,
        argPackAdvance: F,
        readValueFromPointer: i
      }, {
        ignoreDuplicateRegistrations: !0
      });
    }, ot = (r, e, t, n) => {
      if (!(n > 0)) return 0;
      for (var a = t, i = t + n - 1, o = 0; o < r.length; ++o) {
        var s = r.charCodeAt(o);
        if (s >= 55296 && s <= 57343) {
          var c = r.charCodeAt(++o);
          s = 65536 + ((s & 1023) << 10) | c & 1023;
        }
        if (s <= 127) {
          if (t >= i) break;
          e[t++] = s;
        } else if (s <= 2047) {
          if (t + 1 >= i) break;
          e[t++] = 192 | s >> 6, e[t++] = 128 | s & 63;
        } else if (s <= 65535) {
          if (t + 2 >= i) break;
          e[t++] = 224 | s >> 12, e[t++] = 128 | s >> 6 & 63, e[t++] = 128 | s & 63;
        } else {
          if (t + 3 >= i) break;
          e[t++] = 240 | s >> 18, e[t++] = 128 | s >> 12 & 63, e[t++] = 128 | s >> 6 & 63, e[t++] = 128 | s & 63;
        }
      }
      return e[t] = 0, t - a;
    }, st = (r, e, t) => ot(r, w, e, t), ut = (r) => {
      for (var e = 0, t = 0; t < r.length; ++t) {
        var n = r.charCodeAt(t);
        n <= 127 ? e++ : n <= 2047 ? e += 2 : n >= 55296 && n <= 57343 ? (e += 4, ++t) : e += 3;
      }
      return e;
    }, Br = typeof TextDecoder < "u" ? new TextDecoder() : void 0, ct = function(r) {
      let e = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : 0, t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : NaN;
      for (var n = e + t, a = e; r[a] && !(a >= n); ) ++a;
      if (a - e > 16 && r.buffer && Br)
        return Br.decode(r.subarray(e, a));
      for (var i = ""; e < a; ) {
        var o = r[e++];
        if (!(o & 128)) {
          i += String.fromCharCode(o);
          continue;
        }
        var s = r[e++] & 63;
        if ((o & 224) == 192) {
          i += String.fromCharCode((o & 31) << 6 | s);
          continue;
        }
        var c = r[e++] & 63;
        if ((o & 240) == 224 ? o = (o & 15) << 12 | s << 6 | c : o = (o & 7) << 18 | s << 12 | c << 6 | r[e++] & 63, o < 65536)
          i += String.fromCharCode(o);
        else {
          var f = o - 65536;
          i += String.fromCharCode(55296 | f >> 10, 56320 | f & 1023);
        }
      }
      return i;
    }, ft = (r, e) => r ? ct(w, r, e) : "", vt = (r, e) => {
      e = b(e), E(r, {
        name: e,
        fromWireType(t) {
          for (var n = d[t >> 2], a = t + 4, i, o, s = a, o = 0; o <= n; ++o) {
            var c = a + o;
            if (o == n || w[c] == 0) {
              var f = c - s, v = ft(s, f);
              i === void 0 ? i = v : (i += "\0", i += v), s = c + 1;
            }
          }
          return W(t), i;
        },
        toWireType(t, n) {
          n instanceof ArrayBuffer && (n = new Uint8Array(n));
          var a, i = typeof n == "string";
          i || n instanceof Uint8Array || n instanceof Uint8ClampedArray || n instanceof Int8Array || $("Cannot pass non-string to std::string"), i ? a = ut(n) : a = n.length;
          var o = mr(4 + a + 1), s = o + 4;
          if (d[o >> 2] = a, i)
            st(n, s, a + 1);
          else if (i)
            for (var c = 0; c < a; ++c) {
              var f = n.charCodeAt(c);
              f > 255 && (W(s), $("String has UTF-16 code units that do not fit in 8 bits")), w[s + c] = f;
            }
          else
            for (var c = 0; c < a; ++c)
              w[s + c] = n[c];
          return t !== null && t.push(W, o), o;
        },
        argPackAdvance: F,
        readValueFromPointer: er,
        destructorFunction(t) {
          W(t);
        }
      });
    }, Xr = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0, lt = (r, e) => {
      for (var t = r, n = t >> 1, a = n + e / 2; !(n >= a) && N[n]; ) ++n;
      if (t = n << 1, t - r > 32 && Xr) return Xr.decode(w.subarray(r, t));
      for (var i = "", o = 0; !(o >= e / 2); ++o) {
        var s = j[r + o * 2 >> 1];
        if (s == 0) break;
        i += String.fromCharCode(s);
      }
      return i;
    }, dt = (r, e, t) => {
      var n;
      if ((n = t) !== null && n !== void 0 || (t = 2147483647), t < 2) return 0;
      t -= 2;
      for (var a = e, i = t < r.length * 2 ? t / 2 : r.length, o = 0; o < i; ++o) {
        var s = r.charCodeAt(o);
        j[e >> 1] = s, e += 2;
      }
      return j[e >> 1] = 0, e - a;
    }, _t = (r) => r.length * 2, pt = (r, e) => {
      for (var t = 0, n = ""; !(t >= e / 4); ) {
        var a = D[r + t * 4 >> 2];
        if (a == 0) break;
        if (++t, a >= 65536) {
          var i = a - 65536;
          n += String.fromCharCode(55296 | i >> 10, 56320 | i & 1023);
        } else
          n += String.fromCharCode(a);
      }
      return n;
    }, ht = (r, e, t) => {
      var n;
      if ((n = t) !== null && n !== void 0 || (t = 2147483647), t < 4) return 0;
      for (var a = e, i = a + t - 4, o = 0; o < r.length; ++o) {
        var s = r.charCodeAt(o);
        if (s >= 55296 && s <= 57343) {
          var c = r.charCodeAt(++o);
          s = 65536 + ((s & 1023) << 10) | c & 1023;
        }
        if (D[e >> 2] = s, e += 4, e + 4 > i) break;
      }
      return D[e >> 2] = 0, e - a;
    }, gt = (r) => {
      for (var e = 0, t = 0; t < r.length; ++t) {
        var n = r.charCodeAt(t);
        n >= 55296 && n <= 57343 && ++t, e += 4;
      }
      return e;
    }, mt = (r, e, t) => {
      t = b(t);
      var n, a, i, o;
      e === 2 ? (n = lt, a = dt, o = _t, i = (s) => N[s >> 1]) : e === 4 && (n = pt, a = ht, o = gt, i = (s) => d[s >> 2]), E(r, {
        name: t,
        fromWireType: (s) => {
          for (var c = d[s >> 2], f, v = s + 4, l = 0; l <= c; ++l) {
            var h = s + 4 + l * e;
            if (l == c || i(h) == 0) {
              var T = h - v, S = n(v, T);
              f === void 0 ? f = S : (f += "\0", f += S), v = h + e;
            }
          }
          return W(s), f;
        },
        toWireType: (s, c) => {
          typeof c != "string" && $(`Cannot pass non-string to C++ string type ${t}`);
          var f = o(c), v = mr(4 + f + e);
          return d[v >> 2] = f / e, a(c, v + 4, f + e), s !== null && s.push(W, v), v;
        },
        argPackAdvance: F,
        readValueFromPointer: er,
        destructorFunction(s) {
          W(s);
        }
      });
    }, yt = (r, e, t, n, a, i) => {
      rr[r] = {
        name: b(e),
        rawConstructor: X(t, n),
        rawDestructor: X(a, i),
        fields: []
      };
    }, bt = (r, e, t, n, a, i, o, s, c, f) => {
      rr[r].fields.push({
        fieldName: b(e),
        getterReturnType: t,
        getter: X(n, a),
        getterContext: i,
        setterArgumentType: o,
        setter: X(s, c),
        setterContext: f
      });
    }, wt = (r, e) => {
      e = b(e), E(r, {
        isVoid: !0,
        name: e,
        argPackAdvance: 0,
        fromWireType: () => {
        },
        toWireType: (t, n) => {
        }
      });
    }, Tt = (r, e, t) => w.copyWithin(r, e, e + t), gr = [], $t = (r, e, t, n) => (r = gr[r], e = I.toValue(e), r(null, e, t, n)), At = {}, Et = (r) => {
      var e = At[r];
      return e === void 0 ? b(r) : e;
    }, Lr = () => {
      if (typeof globalThis == "object")
        return globalThis;
      function r(e) {
        e.$$$embind_global$$$ = e;
        var t = typeof $$$embind_global$$$ == "object" && e.$$$embind_global$$$ == e;
        return t || delete e.$$$embind_global$$$, t;
      }
      if (typeof $$$embind_global$$$ == "object" || (typeof global == "object" && r(global) ? $$$embind_global$$$ = global : typeof self == "object" && r(self) && ($$$embind_global$$$ = self), typeof $$$embind_global$$$ == "object"))
        return $$$embind_global$$$;
      throw Error("unable to get global object.");
    }, Rt = (r) => r === 0 ? I.toHandle(Lr()) : (r = Et(r), I.toHandle(Lr()[r])), Ct = (r) => {
      var e = gr.length;
      return gr.push(r), e;
    }, Ft = (r, e) => {
      for (var t = new Array(r), n = 0; n < r; ++n)
        t[n] = Vr(d[e + n * 4 >> 2], "parameter " + n);
      return t;
    }, Pt = Reflect.construct, Wt = (r, e, t) => {
      var n = [], a = r.toWireType(n, t);
      return n.length && (d[e >> 2] = I.toHandle(n)), a;
    }, kt = (r, e, t) => {
      var n = Ft(r, e), a = n.shift();
      r--;
      var i = new Array(r), o = (c, f, v, l) => {
        for (var h = 0, T = 0; T < r; ++T)
          i[T] = n[T].readValueFromPointer(l + h), h += n[T].argPackAdvance;
        var S = t === 1 ? Pt(f, i) : f.apply(c, i);
        return Wt(a, v, S);
      }, s = `methodCaller<(${n.map((c) => c.name).join(", ")}) => ${a.name}>`;
      return Ct(nr(s, o));
    }, Ut = (r) => {
      r > 9 && (P[r + 1] += 1);
    }, St = (r) => {
      var e = I.toValue(r);
      _r(e), hr(r);
    }, xt = () => 2147483648, Mt = (r, e) => Math.ceil(r / e) * e, It = (r) => {
      var e = J.buffer, t = (r - e.byteLength + 65535) / 65536 | 0;
      try {
        return J.grow(t), Fr(), 1;
      } catch {
      }
    }, jt = (r) => {
      var e = w.length;
      r >>>= 0;
      var t = xt();
      if (r > t)
        return !1;
      for (var n = 1; n <= 4; n *= 2) {
        var a = e * (1 + 0.2 / n);
        a = Math.min(a, r + 100663296);
        var i = Math.min(t, Mt(Math.max(r, a), 65536)), o = It(i);
        if (o)
          return !0;
      }
      return !1;
    }, Dt = (r) => r;
    Sr = u.InternalError = class extends Error {
      constructor(e) {
        super(e), this.name = "InternalError";
      }
    }, Se(), jr = u.BindingError = class extends Error {
      constructor(e) {
        super(e), this.name = "BindingError";
      }
    }, je(), Nr = u.UnboundTypeError = ze(Error, "UnboundTypeError");
    var Ot = {
      x: Te,
      y: $e,
      a: Ee,
      l: Re,
      u: Ce,
      M: Fe,
      n: Pe,
      f: Ae,
      I: We,
      T: ke,
      H: Ue,
      P: Me,
      N: Oe,
      R: Ne,
      k: Be,
      A: Le,
      S: tt,
      s: at,
      o: it,
      O: vt,
      z: mt,
      B: yt,
      U: bt,
      Q: wt,
      K: Tt,
      E: $t,
      W: hr,
      G: Rt,
      F: kt,
      C: Ut,
      X: St,
      J: jt,
      D: Lt,
      v: en,
      e: Vt,
      c: Zt,
      m: Xt,
      h: rn,
      i: Qt,
      L: Yt,
      q: Gt,
      g: Ht,
      d: Jt,
      b: qt,
      j: Bt,
      p: Nt,
      w: zt,
      r: tn,
      t: Kt,
      V: Dt
    }, y;
    we();
    var Zr = (r) => (Zr = y._)(r), mr = u._malloc = (r) => (mr = u._malloc = y.aa)(r), W = u._free = (r) => (W = u._free = y.ba)(r), p = (r, e) => (p = y.ca)(r, e), Gr = (r) => (Gr = y.da)(r), Kr = (r) => (Kr = y.ea)(r), qr = () => (qr = y.fa)(), Jr = (r) => (Jr = y.ga)(r), Qr = (r) => (Qr = y.ha)(r), Yr = (r, e, t) => (Yr = y.ia)(r, e, t), zr = (r) => (zr = y.ja)(r);
    function Vt(r, e) {
      var t = m();
      try {
        return _(r)(e);
      } catch (n) {
        if (g(t), n !== n + 0) throw n;
        p(1, 0);
      }
    }
    function Ht(r, e) {
      var t = m();
      try {
        _(r)(e);
      } catch (n) {
        if (g(t), n !== n + 0) throw n;
        p(1, 0);
      }
    }
    function Nt(r, e, t, n, a, i) {
      var o = m();
      try {
        _(r)(e, t, n, a, i);
      } catch (s) {
        if (g(o), s !== s + 0) throw s;
        p(1, 0);
      }
    }
    function Bt(r, e, t, n, a) {
      var i = m();
      try {
        _(r)(e, t, n, a);
      } catch (o) {
        if (g(i), o !== o + 0) throw o;
        p(1, 0);
      }
    }
    function Xt(r, e, t, n) {
      var a = m();
      try {
        return _(r)(e, t, n);
      } catch (i) {
        if (g(a), i !== i + 0) throw i;
        p(1, 0);
      }
    }
    function Lt(r, e, t, n, a) {
      var i = m();
      try {
        return _(r)(e, t, n, a);
      } catch (o) {
        if (g(i), o !== o + 0) throw o;
        p(1, 0);
      }
    }
    function Zt(r, e, t) {
      var n = m();
      try {
        return _(r)(e, t);
      } catch (a) {
        if (g(n), a !== a + 0) throw a;
        p(1, 0);
      }
    }
    function Gt(r) {
      var e = m();
      try {
        _(r)();
      } catch (t) {
        if (g(e), t !== t + 0) throw t;
        p(1, 0);
      }
    }
    function Kt(r, e, t, n, a, i, o, s, c, f, v) {
      var l = m();
      try {
        _(r)(e, t, n, a, i, o, s, c, f, v);
      } catch (h) {
        if (g(l), h !== h + 0) throw h;
        p(1, 0);
      }
    }
    function qt(r, e, t, n) {
      var a = m();
      try {
        _(r)(e, t, n);
      } catch (i) {
        if (g(a), i !== i + 0) throw i;
        p(1, 0);
      }
    }
    function Jt(r, e, t) {
      var n = m();
      try {
        _(r)(e, t);
      } catch (a) {
        if (g(n), a !== a + 0) throw a;
        p(1, 0);
      }
    }
    function Qt(r, e, t, n, a, i) {
      var o = m();
      try {
        return _(r)(e, t, n, a, i);
      } catch (s) {
        if (g(o), s !== s + 0) throw s;
        p(1, 0);
      }
    }
    function Yt(r, e, t, n, a, i, o) {
      var s = m();
      try {
        return _(r)(e, t, n, a, i, o);
      } catch (c) {
        if (g(s), c !== c + 0) throw c;
        p(1, 0);
      }
    }
    function zt(r, e, t, n, a, i, o, s) {
      var c = m();
      try {
        _(r)(e, t, n, a, i, o, s);
      } catch (f) {
        if (g(c), f !== f + 0) throw f;
        p(1, 0);
      }
    }
    function rn(r, e, t, n, a) {
      var i = m();
      try {
        return _(r)(e, t, n, a);
      } catch (o) {
        if (g(i), o !== o + 0) throw o;
        p(1, 0);
      }
    }
    function en(r) {
      var e = m();
      try {
        return _(r)();
      } catch (t) {
        if (g(e), t !== t + 0) throw t;
        p(1, 0);
      }
    }
    function tn(r, e, t, n, a, i, o, s, c) {
      var f = m();
      try {
        _(r)(e, t, n, a, i, o, s, c);
      } catch (v) {
        if (g(f), v !== v + 0) throw v;
        p(1, 0);
      }
    }
    var ir;
    B = function r() {
      ir || re(), ir || (B = r);
    };
    function re() {
      if (x > 0 || (oe(), x > 0))
        return;
      function r() {
        var e;
        ir || (ir = !0, u.calledRun = !0, !Er && (se(), Tr(u), (e = u.onRuntimeInitialized) === null || e === void 0 || e.call(u), ue()));
      }
      u.setStatus ? (u.setStatus("Running..."), setTimeout(() => {
        setTimeout(() => u.setStatus(""), 1), r();
      }, 1)) : r();
    }
    if (u.preInit)
      for (typeof u.preInit == "function" && (u.preInit = [u.preInit]); u.preInit.length > 0; )
        u.preInit.pop()();
    return re(), wr = te, wr;
  };
})();
function vn(k) {
  return an(
    br,
    k
  );
}
function ln(k) {
  return on(
    br,
    k
  );
}
async function dn(k, G) {
  return sn(
    br,
    k,
    G
  );
}
export {
  hn as barcodeFormats,
  gn as characterSets,
  mn as defaultEncodeHints,
  yn as defaultWriterOptions,
  vn as getZXingModule,
  bn as purgeZXingModule,
  ln as setZXingModuleOverrides,
  dn as writeBarcodeToImageFile,
  wn as writeInputEccLevels
};
