// zxing-wasm/dist/es/reader/index.js
import { g as Ei, s as ki, r as Si, a as ji } from "../core-BYUmLXDu.js";
import { b as Bi, c as Ni, e as zi, f as Li, d as Gi, d as Zi, h as Xi, p as qi, i as Ki, t as Qi } from "../core-BYUmLXDu.js";
var Tr = (() => {
  var L;
  var S = typeof document < "u" && ((L = document.currentScript) == null ? void 0 : L.tagName.toUpperCase()) === "SCRIPT" ? document.currentScript.src : void 0;
  return function(xe = {}) {
    var Lr, l = xe, Gr, ir, Me = new Promise((e, r) => {
      Gr = e, ir = r;
    }), Ie = typeof window == "object", He = typeof Bun < "u", Cr = typeof WorkerGlobalScope < "u";
    typeof process == "object" && typeof process.versions == "object" && typeof process.versions.node == "string" && process.type != "renderer";
    var Zr = Object.assign({}, l), Xr = "./this.program", j = "";
    function Ve(e) {
      return l.locateFile ? l.locateFile(e, j) : j + e;
    }
    var qr, Pr;
    if (Ie || Cr || He) {
      var Ar;
      Cr ? j = self.location.href : typeof document < "u" && ((Ar = document.currentScript) === null || Ar === void 0 ? void 0 : Ar.tagName.toUpperCase()) === "SCRIPT" && (j = document.currentScript.src), S && (j = S), j.startsWith("blob:") ? j = "" : j = j.substr(0, j.replace(/[?#].*/, "").lastIndexOf("/") + 1), Cr && (Pr = (e) => {
        var r = new XMLHttpRequest();
        return r.open("GET", e, !1), r.responseType = "arraybuffer", r.send(null), new Uint8Array(r.response);
      }), qr = async (e) => {
        var r = await fetch(e, {
          credentials: "same-origin"
        });
        if (r.ok)
          return r.arrayBuffer();
        throw new Error(r.status + " : " + r.url);
      };
    }
    var Be = l.print || console.log.bind(console), G = l.printErr || console.error.bind(console);
    Object.assign(l, Zr), Zr = null, l.arguments && l.arguments, l.thisProgram && (Xr = l.thisProgram);
    var ar = l.wasmBinary, or, Kr = !1, W, R, Z, K, H, $, Qr, Yr;
    function Jr() {
      var e = or.buffer;
      l.HEAP8 = W = new Int8Array(e), l.HEAP16 = Z = new Int16Array(e), l.HEAPU8 = R = new Uint8Array(e), l.HEAPU16 = K = new Uint16Array(e), l.HEAP32 = H = new Int32Array(e), l.HEAPU32 = $ = new Uint32Array(e), l.HEAPF32 = Qr = new Float32Array(e), l.HEAPF64 = Yr = new Float64Array(e);
    }
    var re = [], ee = [], te = [];
    function Ne() {
      if (l.preRun)
        for (typeof l.preRun == "function" && (l.preRun = [l.preRun]); l.preRun.length; )
          Ge(l.preRun.shift());
      Fr(re);
    }
    function ze() {
      Fr(ee);
    }
    function Le() {
      if (l.postRun)
        for (typeof l.postRun == "function" && (l.postRun = [l.postRun]); l.postRun.length; )
          Xe(l.postRun.shift());
      Fr(te);
    }
    function Ge(e) {
      re.unshift(e);
    }
    function Ze(e) {
      ee.unshift(e);
    }
    function Xe(e) {
      te.unshift(e);
    }
    var V = 0, Q = null;
    function qe(e) {
      var r;
      V++, (r = l.monitorRunDependencies) === null || r === void 0 || r.call(l, V);
    }
    function Ke(e) {
      var r;
      if (V--, (r = l.monitorRunDependencies) === null || r === void 0 || r.call(l, V), V == 0 && Q) {
        var t = Q;
        Q = null, t();
      }
    }
    function Rr(e) {
      var r;
      (r = l.onAbort) === null || r === void 0 || r.call(l, e), e = "Aborted(" + e + ")", G(e), Kr = !0, e += ". Build with -sASSERTIONS for more info.";
      var t = new WebAssembly.RuntimeError(e);
      throw ir(t), t;
    }
    var Qe = "data:application/octet-stream;base64,", ne = (e) => e.startsWith(Qe);
    function Ye() {
      var e = "zxing_reader.wasm";
      return ne(e) ? e : Ve(e);
    }
    var sr;
    function Je(e) {
      if (e == sr && ar)
        return new Uint8Array(ar);
      if (Pr)
        return Pr(e);
      throw "both async and sync fetching of the wasm failed";
    }
    async function rt(e) {
      if (!ar)
        try {
          var r = await qr(e);
          return new Uint8Array(r);
        } catch {
        }
      return Je(e);
    }
    async function et(e, r) {
      try {
        var t = await rt(e), n = await WebAssembly.instantiate(t, r);
        return n;
      } catch (i) {
        G(`failed to asynchronously prepare wasm: ${i}`), Rr(i);
      }
    }
    async function tt(e, r, t) {
      if (!e && typeof WebAssembly.instantiateStreaming == "function" && !ne(r) && typeof fetch == "function")
        try {
          var n = fetch(r, {
            credentials: "same-origin"
          }), i = await WebAssembly.instantiateStreaming(n, t);
          return i;
        } catch (a) {
          G(`wasm streaming compile failed: ${a}`), G("falling back to ArrayBuffer instantiation");
        }
      return et(r, t);
    }
    function nt() {
      return {
        a: Zn
      };
    }
    async function it() {
      var e;
      function r(a, s) {
        return w = a.exports, or = w.za, Jr(), de = w.Da, Ze(w.Aa), Ke(), w;
      }
      qe();
      function t(a) {
        r(a.instance);
      }
      var n = nt();
      if (l.instantiateWasm)
        try {
          return l.instantiateWasm(n, r);
        } catch (a) {
          G(`Module.instantiateWasm callback failed with error: ${a}`), ir(a);
        }
      (e = sr) !== null && e !== void 0 || (sr = Ye());
      try {
        var i = await tt(ar, sr, n);
        return t(i), i;
      } catch (a) {
        ir(a);
        return;
      }
    }
    var Fr = (e) => {
      for (; e.length > 0; )
        e.shift()(l);
    };
    l.noExitRuntime;
    var p = (e) => Pe(e), h = () => Ae(), ur = [], lr = 0, at = (e) => {
      var r = new Er(e);
      return r.get_caught() || (r.set_caught(!0), lr--), r.set_rethrown(!1), ur.push(r), Fe(e), Te(e);
    }, D = 0, ot = () => {
      d(0, 0);
      var e = ur.pop();
      Re(e.excPtr), D = 0;
    };
    class Er {
      constructor(r) {
        this.excPtr = r, this.ptr = r - 24;
      }
      set_type(r) {
        $[this.ptr + 4 >> 2] = r;
      }
      get_type() {
        return $[this.ptr + 4 >> 2];
      }
      set_destructor(r) {
        $[this.ptr + 8 >> 2] = r;
      }
      get_destructor() {
        return $[this.ptr + 8 >> 2];
      }
      set_caught(r) {
        r = r ? 1 : 0, W[this.ptr + 12] = r;
      }
      get_caught() {
        return W[this.ptr + 12] != 0;
      }
      set_rethrown(r) {
        r = r ? 1 : 0, W[this.ptr + 13] = r;
      }
      get_rethrown() {
        return W[this.ptr + 13] != 0;
      }
      init(r, t) {
        this.set_adjusted_ptr(0), this.set_type(r), this.set_destructor(t);
      }
      set_adjusted_ptr(r) {
        $[this.ptr + 16 >> 2] = r;
      }
      get_adjusted_ptr() {
        return $[this.ptr + 16 >> 2];
      }
    }
    var st = (e) => {
      throw D || (D = e), D;
    }, cr = (e) => Ce(e), kr = (e) => {
      var r = D;
      if (!r)
        return cr(0), 0;
      var t = new Er(r);
      t.set_adjusted_ptr(r);
      var n = t.get_type();
      if (!n)
        return cr(0), r;
      for (var i of e) {
        if (i === 0 || i === n)
          break;
        var a = t.ptr + 16;
        if (Ee(i, n, a))
          return cr(i), r;
      }
      return cr(n), r;
    }, ut = () => kr([]), lt = (e) => kr([e]), ct = (e, r) => kr([e, r]), ft = () => {
      var e = ur.pop();
      e || Rr("no exception to throw");
      var r = e.excPtr;
      throw e.get_rethrown() || (ur.push(e), e.set_rethrown(!0), e.set_caught(!1), lr++), D = r, D;
    }, vt = (e, r, t) => {
      var n = new Er(e);
      throw n.init(r, t), D = e, lr++, D;
    }, dt = () => lr, pt = () => Rr(""), fr = {}, Sr = (e) => {
      for (; e.length; ) {
        var r = e.pop(), t = e.pop();
        t(r);
      }
    };
    function Y(e) {
      return this.fromWireType($[e >> 2]);
    }
    var X = {}, B = {}, vr = {}, ie, dr = (e) => {
      throw new ie(e);
    }, N = (e, r, t) => {
      e.forEach((o) => vr[o] = r);
      function n(o) {
        var u = t(o);
        u.length !== e.length && dr("Mismatched type converter count");
        for (var c = 0; c < e.length; ++c)
          E(e[c], u[c]);
      }
      var i = new Array(r.length), a = [], s = 0;
      r.forEach((o, u) => {
        B.hasOwnProperty(o) ? i[u] = B[o] : (a.push(o), X.hasOwnProperty(o) || (X[o] = []), X[o].push(() => {
          i[u] = B[o], ++s, s === a.length && n(i);
        }));
      }), a.length === 0 && n(i);
    }, ht = (e) => {
      var r = fr[e];
      delete fr[e];
      var t = r.rawConstructor, n = r.rawDestructor, i = r.fields, a = i.map((s) => s.getterReturnType).concat(i.map((s) => s.setterArgumentType));
      N([e], a, (s) => {
        var o = {};
        return i.forEach((u, c) => {
          var f = u.fieldName, v = s[c], g = u.getter, m = u.getterContext, b = s[c + i.length], P = u.setter, T = u.setterContext;
          o[f] = {
            read: (C) => v.fromWireType(g(m, C)),
            write: (C, z) => {
              var F = [];
              P(T, C, b.toWireType(F, z)), Sr(F);
            }
          };
        }), [{
          name: r.name,
          fromWireType: (u) => {
            var c = {};
            for (var f in o)
              c[f] = o[f].read(u);
            return n(u), c;
          },
          toWireType: (u, c) => {
            for (var f in o)
              if (!(f in c))
                throw new TypeError(`Missing field: "${f}"`);
            var v = t();
            for (f in o)
              o[f].write(v, c[f]);
            return u !== null && u.push(n, v), v;
          },
          argPackAdvance: O,
          readValueFromPointer: Y,
          destructorFunction: n
        }];
      });
    }, _t = (e, r, t, n, i) => {
    }, gt = () => {
      for (var e = new Array(256), r = 0; r < 256; ++r)
        e[r] = String.fromCharCode(r);
      ae = e;
    }, ae, A = (e) => {
      for (var r = "", t = e; R[t]; )
        r += ae[R[t++]];
      return r;
    }, q, y = (e) => {
      throw new q(e);
    };
    function yt(e, r) {
      let t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : {};
      var n = r.name;
      if (e || y(`type "${n}" must have a positive integer typeid pointer`), B.hasOwnProperty(e)) {
        if (t.ignoreDuplicateRegistrations)
          return;
        y(`Cannot register type '${n}' twice`);
      }
      if (B[e] = r, delete vr[e], X.hasOwnProperty(e)) {
        var i = X[e];
        delete X[e], i.forEach((a) => a());
      }
    }
    function E(e, r) {
      let t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : {};
      return yt(e, r, t);
    }
    var O = 8, mt = (e, r, t, n) => {
      r = A(r), E(e, {
        name: r,
        fromWireType: function(i) {
          return !!i;
        },
        toWireType: function(i, a) {
          return a ? t : n;
        },
        argPackAdvance: O,
        readValueFromPointer: function(i) {
          return this.fromWireType(R[i]);
        },
        destructorFunction: null
      });
    }, $t = (e) => ({
      count: e.count,
      deleteScheduled: e.deleteScheduled,
      preservePointerOnDelete: e.preservePointerOnDelete,
      ptr: e.ptr,
      ptrType: e.ptrType,
      smartPtr: e.smartPtr,
      smartPtrType: e.smartPtrType
    }), jr = (e) => {
      function r(t) {
        return t.$$.ptrType.registeredClass.name;
      }
      y(r(e) + " instance already deleted");
    }, Wr = !1, oe = (e) => {
    }, bt = (e) => {
      e.smartPtr ? e.smartPtrType.rawDestructor(e.smartPtr) : e.ptrType.registeredClass.rawDestructor(e.ptr);
    }, se = (e) => {
      e.count.value -= 1;
      var r = e.count.value === 0;
      r && bt(e);
    }, ue = (e, r, t) => {
      if (r === t)
        return e;
      if (t.baseClass === void 0)
        return null;
      var n = ue(e, r, t.baseClass);
      return n === null ? null : t.downcast(n);
    }, le = {}, wt = {}, Tt = (e, r) => {
      for (r === void 0 && y("ptr should not be undefined"); e.baseClass; )
        r = e.upcast(r), e = e.baseClass;
      return r;
    }, Ct = (e, r) => (r = Tt(e, r), wt[r]), pr = (e, r) => {
      (!r.ptrType || !r.ptr) && dr("makeClassHandle requires ptr and ptrType");
      var t = !!r.smartPtrType, n = !!r.smartPtr;
      return t !== n && dr("Both smartPtrType and smartPtr must be specified"), r.count = {
        value: 1
      }, J(Object.create(e, {
        $$: {
          value: r,
          writable: !0
        }
      }));
    };
    function Pt(e) {
      var r = this.getPointee(e);
      if (!r)
        return this.destructor(e), null;
      var t = Ct(this.registeredClass, r);
      if (t !== void 0) {
        if (t.$$.count.value === 0)
          return t.$$.ptr = r, t.$$.smartPtr = e, t.clone();
        var n = t.clone();
        return this.destructor(e), n;
      }
      function i() {
        return this.isSmartPointer ? pr(this.registeredClass.instancePrototype, {
          ptrType: this.pointeeType,
          ptr: r,
          smartPtrType: this,
          smartPtr: e
        }) : pr(this.registeredClass.instancePrototype, {
          ptrType: this,
          ptr: e
        });
      }
      var a = this.registeredClass.getActualType(r), s = le[a];
      if (!s)
        return i.call(this);
      var o;
      this.isConst ? o = s.constPointerType : o = s.pointerType;
      var u = ue(r, this.registeredClass, o.registeredClass);
      return u === null ? i.call(this) : this.isSmartPointer ? pr(o.registeredClass.instancePrototype, {
        ptrType: o,
        ptr: u,
        smartPtrType: this,
        smartPtr: e
      }) : pr(o.registeredClass.instancePrototype, {
        ptrType: o,
        ptr: u
      });
    }
    var J = (e) => typeof FinalizationRegistry > "u" ? (J = (r) => r, e) : (Wr = new FinalizationRegistry((r) => {
      se(r.$$);
    }), J = (r) => {
      var t = r.$$, n = !!t.smartPtr;
      if (n) {
        var i = {
          $$: t
        };
        Wr.register(r, i, r);
      }
      return r;
    }, oe = (r) => Wr.unregister(r), J(e)), hr = [], At = () => {
      for (; hr.length; ) {
        var e = hr.pop();
        e.$$.deleteScheduled = !1, e.delete();
      }
    }, ce, Rt = () => {
      Object.assign(_r.prototype, {
        isAliasOf(e) {
          if (!(this instanceof _r) || !(e instanceof _r))
            return !1;
          var r = this.$$.ptrType.registeredClass, t = this.$$.ptr;
          e.$$ = e.$$;
          for (var n = e.$$.ptrType.registeredClass, i = e.$$.ptr; r.baseClass; )
            t = r.upcast(t), r = r.baseClass;
          for (; n.baseClass; )
            i = n.upcast(i), n = n.baseClass;
          return r === n && t === i;
        },
        clone() {
          if (this.$$.ptr || jr(this), this.$$.preservePointerOnDelete)
            return this.$$.count.value += 1, this;
          var e = J(Object.create(Object.getPrototypeOf(this), {
            $$: {
              value: $t(this.$$)
            }
          }));
          return e.$$.count.value += 1, e.$$.deleteScheduled = !1, e;
        },
        delete() {
          this.$$.ptr || jr(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && y("Object already scheduled for deletion"), oe(this), se(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
        },
        isDeleted() {
          return !this.$$.ptr;
        },
        deleteLater() {
          return this.$$.ptr || jr(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && y("Object already scheduled for deletion"), hr.push(this), hr.length === 1 && ce && ce(At), this.$$.deleteScheduled = !0, this;
        }
      });
    };
    function _r() {
    }
    var rr = (e, r) => Object.defineProperty(r, "name", {
      value: e
    }), fe = (e, r, t) => {
      if (e[r].overloadTable === void 0) {
        var n = e[r];
        e[r] = function() {
          for (var i = arguments.length, a = new Array(i), s = 0; s < i; s++)
            a[s] = arguments[s];
          return e[r].overloadTable.hasOwnProperty(a.length) || y(`Function '${t}' called with an invalid number of arguments (${a.length}) - expects one of (${e[r].overloadTable})!`), e[r].overloadTable[a.length].apply(this, a);
        }, e[r].overloadTable = [], e[r].overloadTable[n.argCount] = n;
      }
    }, Or = (e, r, t) => {
      l.hasOwnProperty(e) ? ((t === void 0 || l[e].overloadTable !== void 0 && l[e].overloadTable[t] !== void 0) && y(`Cannot register public name '${e}' twice`), fe(l, e, e), l[e].overloadTable.hasOwnProperty(t) && y(`Cannot register multiple overloads of a function with the same number of arguments (${t})!`), l[e].overloadTable[t] = r) : (l[e] = r, l[e].argCount = t);
    }, Ft = 48, Et = 57, kt = (e) => {
      e = e.replace(/[^a-zA-Z0-9_]/g, "$");
      var r = e.charCodeAt(0);
      return r >= Ft && r <= Et ? `_${e}` : e;
    };
    function St(e, r, t, n, i, a, s, o) {
      this.name = e, this.constructor = r, this.instancePrototype = t, this.rawDestructor = n, this.baseClass = i, this.getActualType = a, this.upcast = s, this.downcast = o, this.pureVirtualFunctions = [];
    }
    var Dr = (e, r, t) => {
      for (; r !== t; )
        r.upcast || y(`Expected null or instance of ${t.name}, got an instance of ${r.name}`), e = r.upcast(e), r = r.baseClass;
      return e;
    };
    function jt(e, r) {
      if (r === null)
        return this.isReference && y(`null is not a valid ${this.name}`), 0;
      r.$$ || y(`Cannot pass "${Vr(r)}" as a ${this.name}`), r.$$.ptr || y(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Dr(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function Wt(e, r) {
      var t;
      if (r === null)
        return this.isReference && y(`null is not a valid ${this.name}`), this.isSmartPointer ? (t = this.rawConstructor(), e !== null && e.push(this.rawDestructor, t), t) : 0;
      (!r || !r.$$) && y(`Cannot pass "${Vr(r)}" as a ${this.name}`), r.$$.ptr || y(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && r.$$.ptrType.isConst && y(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
      var n = r.$$.ptrType.registeredClass;
      if (t = Dr(r.$$.ptr, n, this.registeredClass), this.isSmartPointer)
        switch (r.$$.smartPtr === void 0 && y("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            r.$$.smartPtrType === this ? t = r.$$.smartPtr : y(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            t = r.$$.smartPtr;
            break;
          case 2:
            if (r.$$.smartPtrType === this)
              t = r.$$.smartPtr;
            else {
              var i = r.clone();
              t = this.rawShare(t, x.toHandle(() => i.delete())), e !== null && e.push(this.rawDestructor, t);
            }
            break;
          default:
            y("Unsupporting sharing policy");
        }
      return t;
    }
    function Ot(e, r) {
      if (r === null)
        return this.isReference && y(`null is not a valid ${this.name}`), 0;
      r.$$ || y(`Cannot pass "${Vr(r)}" as a ${this.name}`), r.$$.ptr || y(`Cannot pass deleted object as a pointer of type ${this.name}`), r.$$.ptrType.isConst && y(`Cannot convert argument of type ${r.$$.ptrType.name} to parameter type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Dr(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    var Dt = () => {
      Object.assign(gr.prototype, {
        getPointee(e) {
          return this.rawGetPointee && (e = this.rawGetPointee(e)), e;
        },
        destructor(e) {
          var r;
          (r = this.rawDestructor) === null || r === void 0 || r.call(this, e);
        },
        argPackAdvance: O,
        readValueFromPointer: Y,
        fromWireType: Pt
      });
    };
    function gr(e, r, t, n, i, a, s, o, u, c, f) {
      this.name = e, this.registeredClass = r, this.isReference = t, this.isConst = n, this.isSmartPointer = i, this.pointeeType = a, this.sharingPolicy = s, this.rawGetPointee = o, this.rawConstructor = u, this.rawShare = c, this.rawDestructor = f, !i && r.baseClass === void 0 ? n ? (this.toWireType = jt, this.destructorFunction = null) : (this.toWireType = Ot, this.destructorFunction = null) : this.toWireType = Wt;
    }
    var ve = (e, r, t) => {
      l.hasOwnProperty(e) || dr("Replacing nonexistent public symbol"), l[e].overloadTable !== void 0 && t !== void 0 ? l[e].overloadTable[t] = r : (l[e] = r, l[e].argCount = t);
    }, Ut = (e, r, t) => {
      e = e.replace(/p/g, "i");
      var n = l["dynCall_" + e];
      return n(r, ...t);
    }, yr = [], de, _ = (e) => {
      var r = yr[e];
      return r || (e >= yr.length && (yr.length = e + 1), yr[e] = r = de.get(e)), r;
    }, xt = function(e, r) {
      let t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : [];
      if (e.includes("j"))
        return Ut(e, r, t);
      var n = _(r)(...t);
      return n;
    }, Mt = (e, r) => function() {
      for (var t = arguments.length, n = new Array(t), i = 0; i < t; i++)
        n[i] = arguments[i];
      return xt(e, r, n);
    }, k = (e, r) => {
      e = A(e);
      function t() {
        return e.includes("j") ? Mt(e, r) : _(r);
      }
      var n = t();
      return typeof n != "function" && y(`unknown function pointer with signature ${e}: ${r}`), n;
    }, It = (e, r) => {
      var t = rr(r, function(n) {
        this.name = r, this.message = n;
        var i = new Error(n).stack;
        i !== void 0 && (this.stack = this.toString() + `
` + i.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return t.prototype = Object.create(e.prototype), t.prototype.constructor = t, t.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, t;
    }, pe, he = (e) => {
      var r = we(e), t = A(r);
      return M(r), t;
    }, mr = (e, r) => {
      var t = [], n = {};
      function i(a) {
        if (!n[a] && !B[a]) {
          if (vr[a]) {
            vr[a].forEach(i);
            return;
          }
          t.push(a), n[a] = !0;
        }
      }
      throw r.forEach(i), new pe(`${e}: ` + t.map(he).join([", "]));
    }, Ht = (e, r, t, n, i, a, s, o, u, c, f, v, g) => {
      f = A(f), a = k(i, a), o && (o = k(s, o)), c && (c = k(u, c)), g = k(v, g);
      var m = kt(f);
      Or(m, function() {
        mr(`Cannot construct ${f} due to unbound types`, [n]);
      }), N([e, r, t], n ? [n] : [], (b) => {
        b = b[0];
        var P, T;
        n ? (P = b.registeredClass, T = P.instancePrototype) : T = _r.prototype;
        var C = rr(f, function() {
          if (Object.getPrototypeOf(this) !== z)
            throw new q("Use 'new' to construct " + f);
          if (F.constructor_body === void 0)
            throw new q(f + " has no accessible constructor");
          for (var De = arguments.length, br = new Array(De), wr = 0; wr < De; wr++)
            br[wr] = arguments[wr];
          var Ue = F.constructor_body[br.length];
          if (Ue === void 0)
            throw new q(`Tried to invoke ctor of ${f} with invalid number of parameters (${br.length}) - expected (${Object.keys(F.constructor_body).toString()}) parameters instead!`);
          return Ue.apply(this, br);
        }), z = Object.create(T, {
          constructor: {
            value: C
          }
        });
        C.prototype = z;
        var F = new St(f, C, z, g, P, a, o, c);
        if (F.baseClass) {
          var I, nr;
          (nr = (I = F.baseClass).__derivedClasses) !== null && nr !== void 0 || (I.__derivedClasses = []), F.baseClass.__derivedClasses.push(F);
        }
        var Fi = new gr(f, F, !0, !1, !1), We = new gr(f + "*", F, !1, !1, !1), Oe = new gr(f + " const*", F, !1, !0, !1);
        return le[e] = {
          pointerType: We,
          constPointerType: Oe
        }, ve(m, C), [Fi, We, Oe];
      });
    }, Ur = (e, r) => {
      for (var t = [], n = 0; n < e; n++)
        t.push($[r + n * 4 >> 2]);
      return t;
    };
    function Vt(e) {
      for (var r = 1; r < e.length; ++r)
        if (e[r] !== null && e[r].destructorFunction === void 0)
          return !0;
      return !1;
    }
    function xr(e, r, t, n, i, a) {
      var s = r.length;
      s < 2 && y("argTypes array size mismatch! Must at least get return value and 'this' types!");
      var o = r[1] !== null && t !== null, u = Vt(r), c = r[0].name !== "void", f = s - 2, v = new Array(f), g = [], m = [], b = function() {
        m.length = 0;
        var P;
        g.length = o ? 2 : 1, g[0] = i, o && (P = r[1].toWireType(m, this), g[1] = P);
        for (var T = 0; T < f; ++T)
          v[T] = r[T + 2].toWireType(m, T < 0 || arguments.length <= T ? void 0 : arguments[T]), g.push(v[T]);
        var C = n(...g);
        function z(F) {
          if (u)
            Sr(m);
          else
            for (var I = o ? 1 : 2; I < r.length; I++) {
              var nr = I === 1 ? P : v[I - 2];
              r[I].destructorFunction !== null && r[I].destructorFunction(nr);
            }
          if (c)
            return r[0].fromWireType(F);
        }
        return z(C);
      };
      return rr(e, b);
    }
    var Bt = (e, r, t, n, i, a) => {
      var s = Ur(r, t);
      i = k(n, i), N([], [e], (o) => {
        o = o[0];
        var u = `constructor ${o.name}`;
        if (o.registeredClass.constructor_body === void 0 && (o.registeredClass.constructor_body = []), o.registeredClass.constructor_body[r - 1] !== void 0)
          throw new q(`Cannot register multiple constructors with identical number of parameters (${r - 1}) for class '${o.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return o.registeredClass.constructor_body[r - 1] = () => {
          mr(`Cannot construct ${o.name} due to unbound types`, s);
        }, N([], s, (c) => (c.splice(1, 0, null), o.registeredClass.constructor_body[r - 1] = xr(u, c, null, i, a), [])), [];
      });
    }, _e = (e) => {
      e = e.trim();
      const r = e.indexOf("(");
      return r !== -1 ? e.substr(0, r) : e;
    }, Nt = (e, r, t, n, i, a, s, o, u, c) => {
      var f = Ur(t, n);
      r = A(r), r = _e(r), a = k(i, a), N([], [e], (v) => {
        v = v[0];
        var g = `${v.name}.${r}`;
        r.startsWith("@@") && (r = Symbol[r.substring(2)]), o && v.registeredClass.pureVirtualFunctions.push(r);
        function m() {
          mr(`Cannot call ${g} due to unbound types`, f);
        }
        var b = v.registeredClass.instancePrototype, P = b[r];
        return P === void 0 || P.overloadTable === void 0 && P.className !== v.name && P.argCount === t - 2 ? (m.argCount = t - 2, m.className = v.name, b[r] = m) : (fe(b, r, g), b[r].overloadTable[t - 2] = m), N([], f, (T) => {
          var C = xr(g, T, v, a, s);
          return b[r].overloadTable === void 0 ? (C.argCount = t - 2, b[r] = C) : b[r].overloadTable[t - 2] = C, [];
        }), [];
      });
    }, Mr = [], U = [], Ir = (e) => {
      e > 9 && --U[e + 1] === 0 && (U[e] = void 0, Mr.push(e));
    }, zt = () => U.length / 2 - 5 - Mr.length, Lt = () => {
      U.push(0, 1, void 0, 1, null, 1, !0, 1, !1, 1), l.count_emval_handles = zt;
    }, x = {
      toValue: (e) => (e || y("Cannot use deleted val. handle = " + e), U[e]),
      toHandle: (e) => {
        switch (e) {
          case void 0:
            return 2;
          case null:
            return 4;
          case !0:
            return 6;
          case !1:
            return 8;
          default: {
            const r = Mr.pop() || U.length;
            return U[r] = e, U[r + 1] = 1, r;
          }
        }
      }
    }, ge = {
      name: "emscripten::val",
      fromWireType: (e) => {
        var r = x.toValue(e);
        return Ir(e), r;
      },
      toWireType: (e, r) => x.toHandle(r),
      argPackAdvance: O,
      readValueFromPointer: Y,
      destructorFunction: null
    }, Gt = (e) => E(e, ge), Zt = (e, r, t) => {
      switch (r) {
        case 1:
          return t ? function(n) {
            return this.fromWireType(W[n]);
          } : function(n) {
            return this.fromWireType(R[n]);
          };
        case 2:
          return t ? function(n) {
            return this.fromWireType(Z[n >> 1]);
          } : function(n) {
            return this.fromWireType(K[n >> 1]);
          };
        case 4:
          return t ? function(n) {
            return this.fromWireType(H[n >> 2]);
          } : function(n) {
            return this.fromWireType($[n >> 2]);
          };
        default:
          throw new TypeError(`invalid integer width (${r}): ${e}`);
      }
    }, Xt = (e, r, t, n) => {
      r = A(r);
      function i() {
      }
      i.values = {}, E(e, {
        name: r,
        constructor: i,
        fromWireType: function(a) {
          return this.constructor.values[a];
        },
        toWireType: (a, s) => s.value,
        argPackAdvance: O,
        readValueFromPointer: Zt(r, t, n),
        destructorFunction: null
      }), Or(r, i);
    }, Hr = (e, r) => {
      var t = B[e];
      return t === void 0 && y(`${r} has unknown type ${he(e)}`), t;
    }, qt = (e, r, t) => {
      var n = Hr(e, "enum");
      r = A(r);
      var i = n.constructor, a = Object.create(n.constructor.prototype, {
        value: {
          value: t
        },
        constructor: {
          value: rr(`${n.name}_${r}`, function() {
          })
        }
      });
      i.values[t] = a, i[r] = a;
    }, Vr = (e) => {
      if (e === null)
        return "null";
      var r = typeof e;
      return r === "object" || r === "array" || r === "function" ? e.toString() : "" + e;
    }, Kt = (e, r) => {
      switch (r) {
        case 4:
          return function(t) {
            return this.fromWireType(Qr[t >> 2]);
          };
        case 8:
          return function(t) {
            return this.fromWireType(Yr[t >> 3]);
          };
        default:
          throw new TypeError(`invalid float width (${r}): ${e}`);
      }
    }, Qt = (e, r, t) => {
      r = A(r), E(e, {
        name: r,
        fromWireType: (n) => n,
        toWireType: (n, i) => i,
        argPackAdvance: O,
        readValueFromPointer: Kt(r, t),
        destructorFunction: null
      });
    }, Yt = (e, r, t, n, i, a, s, o) => {
      var u = Ur(r, t);
      e = A(e), e = _e(e), i = k(n, i), Or(e, function() {
        mr(`Cannot call ${e} due to unbound types`, u);
      }, r - 1), N([], u, (c) => {
        var f = [c[0], null].concat(c.slice(1));
        return ve(e, xr(e, f, null, i, a), r - 1), [];
      });
    }, Jt = (e, r, t) => {
      switch (r) {
        case 1:
          return t ? (n) => W[n] : (n) => R[n];
        case 2:
          return t ? (n) => Z[n >> 1] : (n) => K[n >> 1];
        case 4:
          return t ? (n) => H[n >> 2] : (n) => $[n >> 2];
        default:
          throw new TypeError(`invalid integer width (${r}): ${e}`);
      }
    }, rn = (e, r, t, n, i) => {
      r = A(r);
      var a = (f) => f;
      if (n === 0) {
        var s = 32 - 8 * t;
        a = (f) => f << s >>> s;
      }
      var o = r.includes("unsigned"), u = (f, v) => {
      }, c;
      o ? c = function(f, v) {
        return u(v, this.name), v >>> 0;
      } : c = function(f, v) {
        return u(v, this.name), v;
      }, E(e, {
        name: r,
        fromWireType: a,
        toWireType: c,
        argPackAdvance: O,
        readValueFromPointer: Jt(r, t, n !== 0),
        destructorFunction: null
      });
    }, en = (e, r, t) => {
      var n = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], i = n[r];
      function a(s) {
        var o = $[s >> 2], u = $[s + 4 >> 2];
        return new i(W.buffer, u, o);
      }
      t = A(t), E(e, {
        name: t,
        fromWireType: a,
        argPackAdvance: O,
        readValueFromPointer: a
      }, {
        ignoreDuplicateRegistrations: !0
      });
    }, tn = Object.assign({
      optional: !0
    }, ge), nn = (e, r) => {
      E(e, tn);
    }, an = (e, r, t, n) => {
      if (!(n > 0)) return 0;
      for (var i = t, a = t + n - 1, s = 0; s < e.length; ++s) {
        var o = e.charCodeAt(s);
        if (o >= 55296 && o <= 57343) {
          var u = e.charCodeAt(++s);
          o = 65536 + ((o & 1023) << 10) | u & 1023;
        }
        if (o <= 127) {
          if (t >= a) break;
          r[t++] = o;
        } else if (o <= 2047) {
          if (t + 1 >= a) break;
          r[t++] = 192 | o >> 6, r[t++] = 128 | o & 63;
        } else if (o <= 65535) {
          if (t + 2 >= a) break;
          r[t++] = 224 | o >> 12, r[t++] = 128 | o >> 6 & 63, r[t++] = 128 | o & 63;
        } else {
          if (t + 3 >= a) break;
          r[t++] = 240 | o >> 18, r[t++] = 128 | o >> 12 & 63, r[t++] = 128 | o >> 6 & 63, r[t++] = 128 | o & 63;
        }
      }
      return r[t] = 0, t - i;
    }, er = (e, r, t) => an(e, R, r, t), on = (e) => {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n <= 127 ? r++ : n <= 2047 ? r += 2 : n >= 55296 && n <= 57343 ? (r += 4, ++t) : r += 3;
      }
      return r;
    }, ye = typeof TextDecoder < "u" ? new TextDecoder() : void 0, me = function(e) {
      let r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : 0, t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : NaN;
      for (var n = r + t, i = r; e[i] && !(i >= n); ) ++i;
      if (i - r > 16 && e.buffer && ye)
        return ye.decode(e.subarray(r, i));
      for (var a = ""; r < i; ) {
        var s = e[r++];
        if (!(s & 128)) {
          a += String.fromCharCode(s);
          continue;
        }
        var o = e[r++] & 63;
        if ((s & 224) == 192) {
          a += String.fromCharCode((s & 31) << 6 | o);
          continue;
        }
        var u = e[r++] & 63;
        if ((s & 240) == 224 ? s = (s & 15) << 12 | o << 6 | u : s = (s & 7) << 18 | o << 12 | u << 6 | e[r++] & 63, s < 65536)
          a += String.fromCharCode(s);
        else {
          var c = s - 65536;
          a += String.fromCharCode(55296 | c >> 10, 56320 | c & 1023);
        }
      }
      return a;
    }, sn = (e, r) => e ? me(R, e, r) : "", un = (e, r) => {
      r = A(r), E(e, {
        name: r,
        fromWireType(t) {
          for (var n = $[t >> 2], i = t + 4, a, s, o = i, s = 0; s <= n; ++s) {
            var u = i + s;
            if (s == n || R[u] == 0) {
              var c = u - o, f = sn(o, c);
              a === void 0 ? a = f : (a += "\0", a += f), o = u + 1;
            }
          }
          return M(t), a;
        },
        toWireType(t, n) {
          n instanceof ArrayBuffer && (n = new Uint8Array(n));
          var i, a = typeof n == "string";
          a || n instanceof Uint8Array || n instanceof Uint8ClampedArray || n instanceof Int8Array || y("Cannot pass non-string to std::string"), a ? i = on(n) : i = n.length;
          var s = zr(4 + i + 1), o = s + 4;
          if ($[s >> 2] = i, a)
            er(n, o, i + 1);
          else if (a)
            for (var u = 0; u < i; ++u) {
              var c = n.charCodeAt(u);
              c > 255 && (M(o), y("String has UTF-16 code units that do not fit in 8 bits")), R[o + u] = c;
            }
          else
            for (var u = 0; u < i; ++u)
              R[o + u] = n[u];
          return t !== null && t.push(M, s), s;
        },
        argPackAdvance: O,
        readValueFromPointer: Y,
        destructorFunction(t) {
          M(t);
        }
      });
    }, $e = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0, ln = (e, r) => {
      for (var t = e, n = t >> 1, i = n + r / 2; !(n >= i) && K[n]; ) ++n;
      if (t = n << 1, t - e > 32 && $e) return $e.decode(R.subarray(e, t));
      for (var a = "", s = 0; !(s >= r / 2); ++s) {
        var o = Z[e + s * 2 >> 1];
        if (o == 0) break;
        a += String.fromCharCode(o);
      }
      return a;
    }, cn = (e, r, t) => {
      var n;
      if ((n = t) !== null && n !== void 0 || (t = 2147483647), t < 2) return 0;
      t -= 2;
      for (var i = r, a = t < e.length * 2 ? t / 2 : e.length, s = 0; s < a; ++s) {
        var o = e.charCodeAt(s);
        Z[r >> 1] = o, r += 2;
      }
      return Z[r >> 1] = 0, r - i;
    }, fn = (e) => e.length * 2, vn = (e, r) => {
      for (var t = 0, n = ""; !(t >= r / 4); ) {
        var i = H[e + t * 4 >> 2];
        if (i == 0) break;
        if (++t, i >= 65536) {
          var a = i - 65536;
          n += String.fromCharCode(55296 | a >> 10, 56320 | a & 1023);
        } else
          n += String.fromCharCode(i);
      }
      return n;
    }, dn = (e, r, t) => {
      var n;
      if ((n = t) !== null && n !== void 0 || (t = 2147483647), t < 4) return 0;
      for (var i = r, a = i + t - 4, s = 0; s < e.length; ++s) {
        var o = e.charCodeAt(s);
        if (o >= 55296 && o <= 57343) {
          var u = e.charCodeAt(++s);
          o = 65536 + ((o & 1023) << 10) | u & 1023;
        }
        if (H[r >> 2] = o, r += 4, r + 4 > a) break;
      }
      return H[r >> 2] = 0, r - i;
    }, pn = (e) => {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n >= 55296 && n <= 57343 && ++t, r += 4;
      }
      return r;
    }, hn = (e, r, t) => {
      t = A(t);
      var n, i, a, s;
      r === 2 ? (n = ln, i = cn, s = fn, a = (o) => K[o >> 1]) : r === 4 && (n = vn, i = dn, s = pn, a = (o) => $[o >> 2]), E(e, {
        name: t,
        fromWireType: (o) => {
          for (var u = $[o >> 2], c, f = o + 4, v = 0; v <= u; ++v) {
            var g = o + 4 + v * r;
            if (v == u || a(g) == 0) {
              var m = g - f, b = n(f, m);
              c === void 0 ? c = b : (c += "\0", c += b), f = g + r;
            }
          }
          return M(o), c;
        },
        toWireType: (o, u) => {
          typeof u != "string" && y(`Cannot pass non-string to C++ string type ${t}`);
          var c = s(u), f = zr(4 + c + r);
          return $[f >> 2] = c / r, i(u, f + 4, c + r), o !== null && o.push(M, f), f;
        },
        argPackAdvance: O,
        readValueFromPointer: Y,
        destructorFunction(o) {
          M(o);
        }
      });
    }, _n = (e, r, t, n, i, a) => {
      fr[e] = {
        name: A(r),
        rawConstructor: k(t, n),
        rawDestructor: k(i, a),
        fields: []
      };
    }, gn = (e, r, t, n, i, a, s, o, u, c) => {
      fr[e].fields.push({
        fieldName: A(r),
        getterReturnType: t,
        getter: k(n, i),
        getterContext: a,
        setterArgumentType: s,
        setter: k(o, u),
        setterContext: c
      });
    }, yn = (e, r) => {
      r = A(r), E(e, {
        isVoid: !0,
        name: r,
        argPackAdvance: 0,
        fromWireType: () => {
        },
        toWireType: (t, n) => {
        }
      });
    }, mn = (e, r, t) => R.copyWithin(e, r, r + t), Br = [], $n = (e, r, t, n) => (e = Br[e], r = x.toValue(r), e(null, r, t, n)), bn = {}, wn = (e) => {
      var r = bn[e];
      return r === void 0 ? A(e) : r;
    }, be = () => {
      if (typeof globalThis == "object")
        return globalThis;
      function e(r) {
        r.$$$embind_global$$$ = r;
        var t = typeof $$$embind_global$$$ == "object" && r.$$$embind_global$$$ == r;
        return t || delete r.$$$embind_global$$$, t;
      }
      if (typeof $$$embind_global$$$ == "object" || (typeof global == "object" && e(global) ? $$$embind_global$$$ = global : typeof self == "object" && e(self) && ($$$embind_global$$$ = self), typeof $$$embind_global$$$ == "object"))
        return $$$embind_global$$$;
      throw Error("unable to get global object.");
    }, Tn = (e) => e === 0 ? x.toHandle(be()) : (e = wn(e), x.toHandle(be()[e])), Cn = (e) => {
      var r = Br.length;
      return Br.push(e), r;
    }, Pn = (e, r) => {
      for (var t = new Array(e), n = 0; n < e; ++n)
        t[n] = Hr($[r + n * 4 >> 2], "parameter " + n);
      return t;
    }, An = Reflect.construct, Rn = (e, r, t) => {
      var n = [], i = e.toWireType(n, t);
      return n.length && ($[r >> 2] = x.toHandle(n)), i;
    }, Fn = (e, r, t) => {
      var n = Pn(e, r), i = n.shift();
      e--;
      var a = new Array(e), s = (u, c, f, v) => {
        for (var g = 0, m = 0; m < e; ++m)
          a[m] = n[m].readValueFromPointer(v + g), g += n[m].argPackAdvance;
        var b = t === 1 ? An(c, a) : c.apply(u, a);
        return Rn(i, f, b);
      }, o = `methodCaller<(${n.map((u) => u.name).join(", ")}) => ${i.name}>`;
      return Cn(rr(o, s));
    }, En = (e) => {
      e > 9 && (U[e + 1] += 1);
    }, kn = (e) => {
      var r = x.toValue(e);
      Sr(r), Ir(e);
    }, Sn = (e, r) => {
      e = Hr(e, "_emval_take_value");
      var t = e.readValueFromPointer(r);
      return x.toHandle(t);
    }, jn = (e, r, t, n) => {
      var i = (/* @__PURE__ */ new Date()).getFullYear(), a = new Date(i, 0, 1), s = new Date(i, 6, 1), o = a.getTimezoneOffset(), u = s.getTimezoneOffset(), c = Math.max(o, u);
      $[e >> 2] = c * 60, H[r >> 2] = +(o != u);
      var f = (m) => {
        var b = m >= 0 ? "-" : "+", P = Math.abs(m), T = String(Math.floor(P / 60)).padStart(2, "0"), C = String(P % 60).padStart(2, "0");
        return `UTC${b}${T}${C}`;
      }, v = f(o), g = f(u);
      u < o ? (er(v, t, 17), er(g, n, 17)) : (er(v, n, 17), er(g, t, 17));
    }, Wn = () => 2147483648, On = (e, r) => Math.ceil(e / r) * r, Dn = (e) => {
      var r = or.buffer, t = (e - r.byteLength + 65535) / 65536 | 0;
      try {
        return or.grow(t), Jr(), 1;
      } catch {
      }
    }, Un = (e) => {
      var r = R.length;
      e >>>= 0;
      var t = Wn();
      if (e > t)
        return !1;
      for (var n = 1; n <= 4; n *= 2) {
        var i = r * (1 + 0.2 / n);
        i = Math.min(i, e + 100663296);
        var a = Math.min(t, On(Math.max(e, i), 65536)), s = Dn(a);
        if (s)
          return !0;
      }
      return !1;
    }, Nr = {}, xn = () => Xr || "./this.program", tr = () => {
      if (!tr.strings) {
        var e = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", r = {
          USER: "web_user",
          LOGNAME: "web_user",
          PATH: "/",
          PWD: "/",
          HOME: "/home/web_user",
          LANG: e,
          _: xn()
        };
        for (var t in Nr)
          Nr[t] === void 0 ? delete r[t] : r[t] = Nr[t];
        var n = [];
        for (var t in r)
          n.push(`${t}=${r[t]}`);
        tr.strings = n;
      }
      return tr.strings;
    }, Mn = (e, r) => {
      for (var t = 0; t < e.length; ++t)
        W[r++] = e.charCodeAt(t);
      W[r] = 0;
    }, In = (e, r) => {
      var t = 0;
      return tr().forEach((n, i) => {
        var a = r + t;
        $[e + i * 4 >> 2] = a, Mn(n, a), t += n.length + 1;
      }), 0;
    }, Hn = (e, r) => {
      var t = tr();
      $[e >> 2] = t.length;
      var n = 0;
      return t.forEach((i) => n += i.length + 1), $[r >> 2] = n, 0;
    }, Vn = (e) => 52;
    function Bn(e, r, t, n, i) {
      return 70;
    }
    var Nn = [null, [], []], zn = (e, r) => {
      var t = Nn[e];
      r === 0 || r === 10 ? ((e === 1 ? Be : G)(me(t)), t.length = 0) : t.push(r);
    }, Ln = (e, r, t, n) => {
      for (var i = 0, a = 0; a < t; a++) {
        var s = $[r >> 2], o = $[r + 4 >> 2];
        r += 8;
        for (var u = 0; u < o; u++)
          zn(e, R[s + u]);
        i += o;
      }
      return $[n >> 2] = i, 0;
    }, Gn = (e) => e;
    ie = l.InternalError = class extends Error {
      constructor(r) {
        super(r), this.name = "InternalError";
      }
    }, gt(), q = l.BindingError = class extends Error {
      constructor(r) {
        super(r), this.name = "BindingError";
      }
    }, Rt(), Dt(), pe = l.UnboundTypeError = It(Error, "UnboundTypeError"), Lt();
    var Zn = {
      s: at,
      w: ot,
      a: ut,
      j: lt,
      l: ct,
      O: ft,
      q: vt,
      fa: dt,
      d: st,
      ba: pt,
      va: ht,
      aa: _t,
      pa: mt,
      ta: Ht,
      sa: Bt,
      E: Nt,
      na: Gt,
      F: Xt,
      n: qt,
      V: Qt,
      W: Yt,
      y: rn,
      u: en,
      ua: nn,
      oa: un,
      P: hn,
      L: _n,
      wa: gn,
      qa: yn,
      ia: mn,
      T: $n,
      xa: Ir,
      ya: Tn,
      U: Fn,
      X: En,
      Y: kn,
      ra: Sn,
      ca: jn,
      ga: Un,
      da: In,
      ea: Hn,
      ha: Vn,
      _: Bn,
      S: Ln,
      J: pi,
      C: _i,
      Q: Yn,
      R: Ti,
      r: ci,
      b: Xn,
      D: di,
      ka: yi,
      c: ri,
      ja: mi,
      h: Qn,
      i: ii,
      t: ai,
      N: vi,
      x: si,
      I: bi,
      K: fi,
      A: gi,
      H: Ci,
      $: Ai,
      Z: Ri,
      k: ei,
      f: Jn,
      e: Kn,
      g: qn,
      M: wi,
      m: ni,
      la: hi,
      p: oi,
      v: ui,
      ma: li,
      B: $i,
      o: ti,
      G: Pi,
      z: Gn
    }, w;
    it();
    var we = (e) => (we = w.Ba)(e), M = l._free = (e) => (M = l._free = w.Ca)(e), zr = l._malloc = (e) => (zr = l._malloc = w.Ea)(e), Te = (e) => (Te = w.Fa)(e), d = (e, r) => (d = w.Ga)(e, r), Ce = (e) => (Ce = w.Ha)(e), Pe = (e) => (Pe = w.Ia)(e), Ae = () => (Ae = w.Ja)(), Re = (e) => (Re = w.Ka)(e), Fe = (e) => (Fe = w.La)(e), Ee = (e, r, t) => (Ee = w.Ma)(e, r, t);
    l.dynCall_viijii = (e, r, t, n, i, a, s) => (l.dynCall_viijii = w.Na)(e, r, t, n, i, a, s);
    var ke = l.dynCall_jiii = (e, r, t, n) => (ke = l.dynCall_jiii = w.Oa)(e, r, t, n);
    l.dynCall_jiji = (e, r, t, n, i) => (l.dynCall_jiji = w.Pa)(e, r, t, n, i);
    var Se = l.dynCall_jiiii = (e, r, t, n, i) => (Se = l.dynCall_jiiii = w.Qa)(e, r, t, n, i);
    l.dynCall_iiiiij = (e, r, t, n, i, a, s) => (l.dynCall_iiiiij = w.Ra)(e, r, t, n, i, a, s), l.dynCall_iiiiijj = (e, r, t, n, i, a, s, o, u) => (l.dynCall_iiiiijj = w.Sa)(e, r, t, n, i, a, s, o, u), l.dynCall_iiiiiijj = (e, r, t, n, i, a, s, o, u, c) => (l.dynCall_iiiiiijj = w.Ta)(e, r, t, n, i, a, s, o, u, c);
    function Xn(e, r) {
      var t = h();
      try {
        return _(e)(r);
      } catch (n) {
        if (p(t), n !== n + 0) throw n;
        d(1, 0);
      }
    }
    function qn(e, r, t, n) {
      var i = h();
      try {
        _(e)(r, t, n);
      } catch (a) {
        if (p(i), a !== a + 0) throw a;
        d(1, 0);
      }
    }
    function Kn(e, r, t) {
      var n = h();
      try {
        _(e)(r, t);
      } catch (i) {
        if (p(n), i !== i + 0) throw i;
        d(1, 0);
      }
    }
    function Qn(e, r, t, n) {
      var i = h();
      try {
        return _(e)(r, t, n);
      } catch (a) {
        if (p(i), a !== a + 0) throw a;
        d(1, 0);
      }
    }
    function Yn(e, r, t, n, i) {
      var a = h();
      try {
        return _(e)(r, t, n, i);
      } catch (s) {
        if (p(a), s !== s + 0) throw s;
        d(1, 0);
      }
    }
    function Jn(e, r) {
      var t = h();
      try {
        _(e)(r);
      } catch (n) {
        if (p(t), n !== n + 0) throw n;
        d(1, 0);
      }
    }
    function ri(e, r, t) {
      var n = h();
      try {
        return _(e)(r, t);
      } catch (i) {
        if (p(n), i !== i + 0) throw i;
        d(1, 0);
      }
    }
    function ei(e) {
      var r = h();
      try {
        _(e)();
      } catch (t) {
        if (p(r), t !== t + 0) throw t;
        d(1, 0);
      }
    }
    function ti(e, r, t, n, i, a, s, o, u, c, f) {
      var v = h();
      try {
        _(e)(r, t, n, i, a, s, o, u, c, f);
      } catch (g) {
        if (p(v), g !== g + 0) throw g;
        d(1, 0);
      }
    }
    function ni(e, r, t, n, i) {
      var a = h();
      try {
        _(e)(r, t, n, i);
      } catch (s) {
        if (p(a), s !== s + 0) throw s;
        d(1, 0);
      }
    }
    function ii(e, r, t, n, i) {
      var a = h();
      try {
        return _(e)(r, t, n, i);
      } catch (s) {
        if (p(a), s !== s + 0) throw s;
        d(1, 0);
      }
    }
    function ai(e, r, t, n, i, a) {
      var s = h();
      try {
        return _(e)(r, t, n, i, a);
      } catch (o) {
        if (p(s), o !== o + 0) throw o;
        d(1, 0);
      }
    }
    function oi(e, r, t, n, i, a) {
      var s = h();
      try {
        _(e)(r, t, n, i, a);
      } catch (o) {
        if (p(s), o !== o + 0) throw o;
        d(1, 0);
      }
    }
    function si(e, r, t, n, i, a, s) {
      var o = h();
      try {
        return _(e)(r, t, n, i, a, s);
      } catch (u) {
        if (p(o), u !== u + 0) throw u;
        d(1, 0);
      }
    }
    function ui(e, r, t, n, i, a, s, o) {
      var u = h();
      try {
        _(e)(r, t, n, i, a, s, o);
      } catch (c) {
        if (p(u), c !== c + 0) throw c;
        d(1, 0);
      }
    }
    function li(e, r, t, n, i, a, s, o, u) {
      var c = h();
      try {
        _(e)(r, t, n, i, a, s, o, u);
      } catch (f) {
        if (p(c), f !== f + 0) throw f;
        d(1, 0);
      }
    }
    function ci(e) {
      var r = h();
      try {
        return _(e)();
      } catch (t) {
        if (p(r), t !== t + 0) throw t;
        d(1, 0);
      }
    }
    function fi(e, r, t, n, i, a, s, o, u) {
      var c = h();
      try {
        return _(e)(r, t, n, i, a, s, o, u);
      } catch (f) {
        if (p(c), f !== f + 0) throw f;
        d(1, 0);
      }
    }
    function vi(e, r, t, n, i, a, s) {
      var o = h();
      try {
        return _(e)(r, t, n, i, a, s);
      } catch (u) {
        if (p(o), u !== u + 0) throw u;
        d(1, 0);
      }
    }
    function di(e, r, t, n) {
      var i = h();
      try {
        return _(e)(r, t, n);
      } catch (a) {
        if (p(i), a !== a + 0) throw a;
        d(1, 0);
      }
    }
    function pi(e, r, t, n) {
      var i = h();
      try {
        return _(e)(r, t, n);
      } catch (a) {
        if (p(i), a !== a + 0) throw a;
        d(1, 0);
      }
    }
    function hi(e, r, t, n, i, a, s, o) {
      var u = h();
      try {
        _(e)(r, t, n, i, a, s, o);
      } catch (c) {
        if (p(u), c !== c + 0) throw c;
        d(1, 0);
      }
    }
    function _i(e, r, t, n, i, a) {
      var s = h();
      try {
        return _(e)(r, t, n, i, a);
      } catch (o) {
        if (p(s), o !== o + 0) throw o;
        d(1, 0);
      }
    }
    function gi(e, r, t, n, i, a, s, o, u, c) {
      var f = h();
      try {
        return _(e)(r, t, n, i, a, s, o, u, c);
      } catch (v) {
        if (p(f), v !== v + 0) throw v;
        d(1, 0);
      }
    }
    function yi(e, r, t) {
      var n = h();
      try {
        return _(e)(r, t);
      } catch (i) {
        if (p(n), i !== i + 0) throw i;
        d(1, 0);
      }
    }
    function mi(e, r, t, n, i) {
      var a = h();
      try {
        return _(e)(r, t, n, i);
      } catch (s) {
        if (p(a), s !== s + 0) throw s;
        d(1, 0);
      }
    }
    function $i(e, r, t, n, i, a, s, o, u, c) {
      var f = h();
      try {
        _(e)(r, t, n, i, a, s, o, u, c);
      } catch (v) {
        if (p(f), v !== v + 0) throw v;
        d(1, 0);
      }
    }
    function bi(e, r, t, n, i, a, s, o) {
      var u = h();
      try {
        return _(e)(r, t, n, i, a, s, o);
      } catch (c) {
        if (p(u), c !== c + 0) throw c;
        d(1, 0);
      }
    }
    function wi(e, r, t, n, i, a, s) {
      var o = h();
      try {
        _(e)(r, t, n, i, a, s);
      } catch (u) {
        if (p(o), u !== u + 0) throw u;
        d(1, 0);
      }
    }
    function Ti(e, r, t, n) {
      var i = h();
      try {
        return _(e)(r, t, n);
      } catch (a) {
        if (p(i), a !== a + 0) throw a;
        d(1, 0);
      }
    }
    function Ci(e, r, t, n, i, a, s, o, u, c, f, v) {
      var g = h();
      try {
        return _(e)(r, t, n, i, a, s, o, u, c, f, v);
      } catch (m) {
        if (p(g), m !== m + 0) throw m;
        d(1, 0);
      }
    }
    function Pi(e, r, t, n, i, a, s, o, u, c, f, v, g, m, b, P) {
      var T = h();
      try {
        _(e)(r, t, n, i, a, s, o, u, c, f, v, g, m, b, P);
      } catch (C) {
        if (p(T), C !== C + 0) throw C;
        d(1, 0);
      }
    }
    function Ai(e, r, t, n) {
      var i = h();
      try {
        return ke(e, r, t, n);
      } catch (a) {
        if (p(i), a !== a + 0) throw a;
        d(1, 0);
      }
    }
    function Ri(e, r, t, n, i) {
      var a = h();
      try {
        return Se(e, r, t, n, i);
      } catch (s) {
        if (p(a), s !== s + 0) throw s;
        d(1, 0);
      }
    }
    var $r;
    Q = function e() {
      $r || je(), $r || (Q = e);
    };
    function je() {
      if (V > 0 || (Ne(), V > 0))
        return;
      function e() {
        var r;
        $r || ($r = !0, l.calledRun = !0, !Kr && (ze(), Gr(l), (r = l.onRuntimeInitialized) === null || r === void 0 || r.call(l), Le()));
      }
      l.setStatus ? (l.setStatus("Running..."), setTimeout(() => {
        setTimeout(() => l.setStatus(""), 1), e();
      }, 1)) : e();
    }
    if (l.preInit)
      for (typeof l.preInit == "function" && (l.preInit = [l.preInit]); l.preInit.length > 0; )
        l.preInit.pop()();
    return je(), Lr = Me, Lr;
  };
})();
function Ui(S) {
  return Ei(
    Tr,
    S
  );
}
function xi(S) {
  return ki(
    Tr,
    S
  );
}
async function Mi(S, L) {
  return Si(
    Tr,
    S,
    L
  );
}
async function Ii(S, L) {
  return ji(
    Tr,
    S,
    L
  );
}
export {
  Bi as barcodeFormats,
  Ni as binarizers,
  zi as characterSets,
  Li as contentTypes,
  Gi as defaultDecodeHints,
  Zi as defaultReaderOptions,
  Xi as eanAddOnSymbols,
  Ui as getZXingModule,
  qi as purgeZXingModule,
  Ii as readBarcodesFromImageData,
  Mi as readBarcodesFromImageFile,
  Ki as readOutputEccLevels,
  xi as setZXingModuleOverrides,
  Qi as textModes
};
